src/controllers/wallet/wallet.controller.js

```diff
@@ -162,6 +162,56 @@ function getTransaction(req, res) {
   }
 }
 
+async function getTxConfirmationBlocks(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const { wallet } = req;
+  const { id } = req.query;
+  // This is O(1) operation, so we can use it to check if this tx belongs to the wallet
+  const tx = wallet.getTx(id);
+  if (!tx) {
+    // We allow only to get data for transactions from the wallet
+    res.send({ success: false, error: `Wallet does not contain transaction with id ${id}` });
+    return;
+  }
+
+  // First get transaction data from full node
+
+  // Disabling this eslint rule because of the way API call is done in the lib
+  // otherwise the code would need to be more complex
+  // We should change this when we refactor the way we call APIs in the lib
+  // (this comment also applies for the getMiningInfo call)
+  // eslint-disable-next-line no-promise-executor-return
+  const txDataResponse = await new Promise(resolve => txApi.getTransaction(id, resolve));
+
+  if (!txDataResponse.success) {
+    res.send({ success: false, error: 'Failed to get transaction data from the full node.' });
+    return;
+  }
+
+  // Now we get the current height of the network
+  // eslint-disable-next-line no-promise-executor-return
+  const networkHeightResponse = await new Promise(resolve => walletApi.getMiningInfo(resolve));
+
+  if (!networkHeightResponse.success) {
+    res.send({ success: false, error: 'Failed to get network heigth from the full node.' });
+    return;
+  }
+
+  let confirmationNumber = 0;
+
+  // first_block_height will be null until a block confirms this transaction
+  if (txDataResponse.meta && txDataResponse.meta.first_block_height) {
+    confirmationNumber = networkHeightResponse.blocks - txDataResponse.meta.first_block_height;
```

Comment:
Just to understand the correlation between these 2 pieces of information.
- `blocks` is the number of blocks, do we exclude voided blocks?
  - If we don't, this value will always be higher than expected.
- Isn't `first_block_height` always 1 higher than the number of valid blocks?
  - It shouldn't be an issue if we expect to return how many blocks confirm the tx, but I just wanted to be sure of the meaning of this api.

Reply:
`blocks` is actually the number of blocks in the best chain, i.e. the network height.

`first_block_height` is the height of the first block that confirms this transaction.